addons/point_of_sale/static/src/js/models.js

```diff
@@ -173,18 +166,129 @@ class PosGlobalState extends PosModel {
     }
 
     async load_server_data(){
-        const loadedModels = await pos_env.services.rpc({
+        const loadedData = await pos_env.services.rpc({
             model: 'pos.session',
             method: 'load_pos_data',
             args: [[odoo.pos_session_id]],
-        })
-        const tmp = {};
-        for (const model of this.models) {
-            if (model.condition ? !model.condition(this) : false) continue;
-            await model.loaded(this, loadedModels[model.model] || [], tmp);
-        }
+        });
+        await this._processData(loadedData);
         return this.after_load_server_data();
     }
+   async _processData(loadedData) {
+        this.version = loadedData['version'];
+        this.company = loadedData['res.company'];
+        this.dp = loadedData['decimal.precision'];
+        this.units = loadedData['uom.uom'];
+        this.units_by_id = loadedData['units_by_id'];
+        this.states = loadedData['res.country.state'];
+        this.countries = loadedData['res.country'];
+        this.langs = loadedData['res.lang'];
+        this.taxes = loadedData['account.tax'];
+        this.taxes_by_id = loadedData['taxes_by_id'];
+        this.pos_session = loadedData['pos.session'];
+        this._loadPosSession();
+        this.config = loadedData['pos.config'];
+        this._loadPoSConfig();
+        this.bills = loadedData['pos.bill'];
+        this.partners = loadedData['res.partner'];
+        this._loadResPartner();
+        this.picking_type = loadedData['stock.picking.type'];
+        this.user = loadedData['res.users'];
+        this.pricelists = loadedData['product.pricelist'];
+        this.default_pricelist = loadedData['default_pricelist'];
+        this.bank_statement = loadedData['account.bank.statement'];
+        this.product_categories = loadedData['product.category'];
```

Comment:
we don't need this assignment anymore. `.categ` for each product is already assigned in the backend.

Note: this is another example of duplication of data. Imagine if the number of loaded products is 100k+, and each is assigned with `.categ` is is an object itself. The data will double with just that.

Reply:
Indeed, that's why we need to only put the id instead of object in the future for the relationships